src/Ui/UiRequest.py

```diff
@@ -292,7 +293,7 @@ def sendHeader(self, status=200, content_type="text/html", noscript=False, allow
             headers["Access-Control-Allow-Headers"] = "Origin, X-Requested-With, Content-Type, Accept, Cookie, Range"
             headers["Access-Control-Allow-Credentials"] = "true"
 
-        if content_type in ("text/plain", "text/html", "text/css", "application/javascript", "application/json", "application/manifest+json"):
+        if content_type.startswith("text") or content_type in ("application/javascript", "application/json", "application/manifest+json"):
```

Comment:
I think text/ would be more correct.

Reply:
You think wrong.

Reply:
Hm. Let me explain my choice. You want to handle `text/*`, i.e. MIME types within `text` section. That's exactly what `startswith("text/")` does.

On the other hand, `startswith("text")` matches MIME sections that start with `text`. Currently, the only section stating with `text` is `text` itself. Even if this works, it's counterintuitive and it might break in case more MIME sections are added in the future.

Reply:
This is the same as subdirectory matching: `file.startswith(dir + "/")` matches iff `file` is inside `dir` whilst `file.startswith(dir)` doesn't work for e.g. `file="1.txt", dir="1"`.

Reply:
An example [content_type.startswith(‘text’)](https://www.iquda.co.uk/embedded-pdf-file-view-plone/) from www.iquda.co.uk. 

> startswith("text") matches MIME sections that start with text

That's correct and so all text should be encoded in UTF-8. 

There will be never a MIME type like textdirdur/imachug for video files! I don't get it why you talk about directories.... 

The `text` MIME type always used on text files and this will never change, so this pull request can't break anything now or in the future.

Reply:
If you don't mind, I mark it as resolved.

Reply:
`text/*` will always be used for text files, that's right. However, it's not guaranteed that something like `text-like/*` isn't added in the future.

As for the directories, that was rather obvious, wasn't it? MIME sections are like directories, i.e. `text/plain` is like a file called `plain` in `text` directory.

Reply:
```
        # Caching
        if status in (200, 206) and content_type.startswith("text"):  # Cache MIME type text/* for 1 week
            headers["Cache-Control"] = "public, max-age=604800"
        if status in (200, 206) and content_type.startswith("image"):  # Cache MIME type image/* for 1 month
            headers["Cache-Control"] = "public, max-age=2629746"
        if status in (200, 206) and content_type.startswith("font"):  # Cache MIME type font/* for 6 months
            headers["Cache-Control"] = "public, max-age=15778476"
        if status in (200, 206) and content_type.startswith("video") or content_type.startswith("audio"):  # Cache MIME type video/* and audio/* for 1 year
            headers["Cache-Control"] = "public, max-age=31556952"
        if status in (200, 206) and content_type.startswith("application") or content_type.startswith("message"):  # Not caching MIME types application/* and message/*
            headers["Cache-Control"] = "no-cache, no-store, private, must-revalidate, max-age=0"
        headers["Content-Type"] = content_type
        headers.update(extra_headers)
        return self.start_response(status_texts[status], list(headers.items()))
```

Reply:
I close this discussion, this is clearly resolved. I can even write `content_type.startswith("te")` and it will still match all text/* MIME types.

Reply:
> This caching pretty much covers the entire mimetypes module.

Right but there's a reason why caching wasn't added...

> So in the future I'm doomed?

Uh, maybe. I don't know whether that's going to happen but there's some probability.

> You talk about directories and how a piece of code look like, but here and now in this pull request we talking about content_type.startswith.

How are MIME types different from file paths? The only difference I can find is that directory names / sections are registered by some committee, not created by users.

Reply:
> I close this discussion, this is clearly resolved. I can even write content_type.startswith("te") and it will still match all text/* MIME types.

It will match text/* but it might match something else like test/* and such. That's clearly not what you're looking for.

Reply:
Right now, it is good without forward slash. When a new MIME arrives like `text-like/*` :rofl: as you said, than I will send another PR. Playing as Nostradamus-Imachug is not the best right now. I can also predict that in the future will nobody use ZeroNet... Right? So why developing it.. You see. I take care of UiRequest when needed.

Reply:
Come on, `text/` will work for sure and `text` might break so you say "I'll fix that later" instead of defeating the problem right now?

Reply:
Let's see you travis... :rofl: